src/pub_verif_token.ts

```diff
@@ -150,57 +216,132 @@ export class TokenResponse {
     }
 }
 
-export function verifyToken(
-    blindRSAMode: BlindRSAMode,
-    token: Token,
-    publicKeyIssuer: CryptoKey,
-): Promise<boolean> {
-    return crypto.subtle.verify(
-        {
-            ...BLIND_RSA.rsaParams,
-            saltLength: blindRSAMode,
-        },
-        publicKeyIssuer,
-        token.authenticator,
-        token.authInput.serialize(),
-    );
-}
+abstract class PubliclyVerifiableIssuer {
+    private suite: (extensions?: Extensions) => Pick<BlindRSA, 'blindSign' | 'verify'>;
 
-export class Issuer {
     constructor(
-        public readonly mode: BlindRSAMode,
+        public readonly mode: BlindRSAMode | PartiallyBlindRSAMode,
         public readonly name: string,
         private readonly privateKey: CryptoKey,
         public readonly publicKey: CryptoKey,
         public readonly params?: BlindRSAPlatformParams,
-    ) {}
+    ) {
+        this.suite = (extensions?: Extensions) => {
+            if (extensions === undefined) {
+                return BLIND_RSA.suite[this.mode]();
```

Comment:
we can avoid this function just by returning the right suite.

```ts
if (!extensions){
  this.suite = BLIND_RSA.suite[this.mode]();
}else{
  const suite = PARTIALLY_BLIND_RSA.suite[this.mode]();
  const serializedExtensions = extensions.serialize();
  this.suite = {
    blindSign: (privateKey: CryptoKey, blindMsg: Uint8Array) =>
                        suite.blindSign(privateKey, blindMsg, serializedExtensions),
    verify: (publicKey: CryptoKey, signature: Uint8Array, message: Uint8Array) =>
                        suite.verify(publicKey, signature, message, serializedExtensions),
  };
}
```

Reply:
you don't know the extensions used for issuance at this point. the goal of this function is to provide the necessary suite function (`blindSign`, `verify`) so they don't need extension when called.

moreover, the issuer receiving the extensions as part of `ExtendedTokenRequest`, I've decided that the extensions would only be known at issuance time, and not upon creating the issuer. This is somehow closer to the spec. If extensions where to be passed at the constructor level, I'd argue it should be `supportedExtensions`, that would not have the value of the extension in them.